File: Swift 4.x/TripServiceKata/Sources/TripServiceKata/UserSession.swift
import Foundation

class UserSession
{
    static var sharedInstance:UserSession = UserSession()
    
    func isUserLoggedIn(_ user:User) throws -> Bool
    {
        throw UnitTestErrorType.dependendClassCallDuringUnitTest
    }
    
    func getLoggedUser() throws -> User?
    {
        throw UnitTestErrorType.dependendClassCallDuringUnitTest
    }
}

File: dart/lib/user_session.dart
import 'package:trip_service/unit_tests_error_type.dart';
import 'package:trip_service/user.dart';

class UserSession {
  UserSession._privateConstructor();

  static final UserSession instance = UserSession._privateConstructor();

  Future<bool> isUserLoggedIn(User user) async {
    throw UnitTestErrorType.dependendClassCallDuringUnitTest;
  }

  Future<User?> getLoggedUser() async {
    throw UnitTestErrorType.dependendClassCallDuringUnitTest;
  }
}
